### PR TITLE
Add docment for TfmdLists

### PR DESCRIPTION
**Pull request by malligaraj**
Implement `TfmdLists` task in issue #8

### CODE REVIEW

File: fastai/data/core.py [hunk 1]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
     def _new(self, items, **kwargs): return super()._new(items, splits=self.splits, **kwargs)
     def subset(self): raise NotImplemented
 
-    def dataloaders(self, bs=64, shuffle_train=None, shuffle=True, val_shuffle=False,n=None, path='.', dl_type=None, dl_kwargs=None,
-                    device=None,drop_last=None,val_bs=None, **kwargs):
+    def dataloaders(self,
+        bs:int=64, # Size of batch
+        shuffle_train:bool=None, # Whether to shuffle training `DataLoader`
```
Comment (kevinbird15) [inline fastai/data/core.py:222]:
```suggestion
        shuffle_train:bool=None, # (Deprecated, use `shuffle`) Whether to shuffle training `DataLoader`
```

File: fastai/data/core.py [hunk 2]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
...
+        val_shuffle:bool=False, # Whether to shuffle validation `DataLoader`
+        n:int=None, # Size of `Datasets` used to create `DataLoader`
+        path:(str, Path)='.', # Path to put `DataLoaders`
+        dl_type=None, # Type of `DataLoader`
+        dl_kwargs=None, # Keyword arguments for individual `DataLoader`s
+        device=None, # Device to put `DataLoaders`
+        drop_last:bool=None, # Whether to drop last incomplete batch
+        val_bs:int=None, # Size of batch for validation `DataLoader`
+        **kwargs
+    ):
```
Comment (kevinbird15) [inline fastai/data/core.py:233]:
```suggestion
    )->DataLoaders:
```

File: fastai/data/core.py [hunk 3]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
...
 
-    def dataloaders(self, bs=64, shuffle_train=None, shuffle=True, val_shuffle=False,n=None, path='.', dl_type=None, dl_kwargs=None,
-                    device=None,drop_last=None,val_bs=None, **kwargs):
+    def dataloaders(self,
+        bs:int=64, # Size of batch
+        shuffle_train:bool=None, # Whether to shuffle training `DataLoader`
+        shuffle:bool=True, # Whether to shuffle training `DataLoader`
+        val_shuffle:bool=False, # Whether to shuffle validation `DataLoader`
+        n:int=None, # Size of `Datasets` used to create `DataLoader`
+        path:(str, Path)='.', # Path to put `DataLoaders`
```
Comment (kevinbird15) [inline fastai/data/core.py:226]:
```suggestion
        path:(str, Path)='.', # Path to put in `DataLoaders`
```

File: fastai/data/core.py [hunk 4]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
...
+        dl_type=None, # Type of `DataLoader`
```
Comment (kevinbird15) [inline fastai/data/core.py:227]:
```suggestion
        dl_type:TfmdDL=None, # Type of `DataLoader`
```

File: fastai/data/core.py [hunk 5]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
...
+        dl_kwargs=None, # Keyword arguments for individual `DataLoader`s
```
Comment (kevinbird15) [inline fastai/data/core.py:228]:
```suggestion
        dl_kwargs:list=None, # list of kwargs to pass to individual `DataLoader`s
```

File: fastai/data/core.py [hunk 6]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
...
+        device=None, # Device to put `DataLoaders`
```
Comment (kevinbird15) [inline fastai/data/core.py:229]:
```suggestion
        device:torch.device=None, # Device to put `DataLoaders`
```

File: fastai/data/core.py [hunk 7]
```diff
@@ -240,8 +252,18 @@ def dataloaders(self, bs=64, shuffle_train=None, shuffle=True, val_shuffle=False
 class TfmdLists(FilteredBase, L, GetAttr):
     "A `Pipeline` of `tfms` applied to a collection of `items`"
     _default='tfms'
-    def __init__(self, items, tfms, use_list=None, do_setup=True, split_idx=None, train_setup=True,
-                 splits=None, types=None, verbose=False, dl_type=None):
+    def __init__(self,
+        items, # List of items to apply `Transform`s to
```
Comment (kevinbird15) [inline fastai/data/core.py:256]:
```suggestion
        items:list, # List of items to apply `Transform`s to
```

File: fastai/data/core.py [hunk 8]
```diff
@@ -240,8 +252,18 @@ def dataloaders(self, bs=64, shuffle_train=None, shuffle=True, val_shuffle=False
...
+        tfms, # List of `Transform`(s) or `Pipeline` to apply
```
Comment (kevinbird15) [inline fastai/data/core.py:257]:
```suggestion
        tfms:(list,Pipeline), # List of `Transform`(s) or `Pipeline` to apply
```

File: fastai/data/core.py [hunk 9]
```diff
@@ -240,8 +252,18 @@ def dataloaders(self, bs=64, shuffle_train=None, shuffle=True, val_shuffle=False
...
+        use_list:bool=None, # Whether to use `list` in `L`
+        do_setup:bool=True, # Whether to call `setup()` for `Transform`
+        split_idx:int=None, # Whether to apply `Transform`(s) to training or validation set. `0` for training set and `1` for validation set
+        train_setup:bool=True, # Whether to apply `Transform`(s) only on training `DataLoader`
+        splits:list=None, # List of indexs for for training and validation sets
+        types=None, # Types of data in `items`
+        verbose:bool=False, # Whether to print verbose output
+        dl_type=None # Type of `DataLoader`
```
Comment (kevinbird15) [inline fastai/data/core.py:265]:
```suggestion
        dl_type:TfmdDL=None # Type of `DataLoader`
```

**Review by kevinbird15 — CHANGES_REQUESTED** [review]
This piece is definitely dense so looking for another reviewer to back me up on some of my changes as well

File: fastai/data/core.py (hunk 2 above)
Comment (malligaraj) [reply fastai/data/core.py:233]: In the [fastai docment Sprint Style Guide](https://github.com/muellerzr/fastai-docment-sprint/blob/master/docs_src/style-guide.md), there seems to be spaces around `->`. May be I can add that in my change?

Comment (kevinbird15) [reply fastai/data/core.py]: Definitely.  Good call on the style guide.  I couldn't remember for sure

File: fastai/data/core.py [hunk 10]
```diff
@@ -217,8 +217,20 @@ def n_subsets(self): return len(self.splits)
...
+        val_shuffle:bool=False, # Whether to shuffle validation `DataLoader`
+        n:int=None, # Size of `Datasets` used to create `DataLoader`
+        path:(str, Path)='.', # Path to put in `DataLoaders`
+        dl_type:TfmdDL=None, # Type of `DataLoader`
+        dl_kwargs:list=None, # list of kwargs to pass to individual `DataLoader`s
+        device:torch.device=None, # Device to put `DataLoaders`
+        drop_last:bool=None, # Whether to drop last incomplete batch
+        val_bs:int=None, # Size of batch for validation `DataLoader`
+        **kwargs
+    ) -> DataLoaders:
```
Comment (warner-benjamin) [inline fastai/data/core.py:233]: Some suggestions to improve conciseness and add info about default setting. Normally, I'd say to remove the "List of" but in `dl_kwargs` case I think could be useful to keep.

```suggestion
    def dataloaders(self,
        bs:int=64, # Batch size
        shuffle_train:bool=None, # (Deprecated, use `shuffle`) Whether to shuffle training `DataLoader`
        shuffle:bool=True, # Shuffle training `DataLoader`
        val_shuffle:bool=False, # Shuffle validation `DataLoader`
        n:int=None, # Size of `Datasets` used to create `DataLoader`
        path:(str, Path)='.', # Path to put in `DataLoaders`
        dl_type:TfmdDL=None, # Type of `DataLoader`
        dl_kwargs:list=None, # List of kwargs to pass to individual `DataLoader`s
        device:torch.device=None, # Device to put `DataLoaders`
        drop_last:bool=None, # Drop last incomplete batch, defaults to `shuffle`
        val_bs:int=None, # Validation batch size, defaults to `bs`
        **kwargs
    ) -> DataLoaders:
```

**Review by warner-benjamin — CHANGES_REQUESTED** [review]
Please replicate suggestions to the notebook and `TfmdLists` method. For most arg strings I think we can remove "List of" as the type hint indicates it's a list.

**Review by kevinbird15 — APPROVED** [review]
Looks good!

**Review by warner-benjamin — APPROVED** [review]
LGTM. The build errors should be fixed now that the main branch uses nbdev git.

**Pull request merged**